lib/rjgit_adapter/git_layer_rjgit.rb

```diff
@@ -201,25 +201,21 @@ def ls_files(query, options = {})
         end
       end
 
-      def revert_path(path, sha1, sha2)
-        patch = @git.diff
+      def revert_path(path, sha1, sha2, ref = 'HEAD^{tree}')
+        patch = generate_patch(sha1, sha2, path)
         return false unless patch
-        if repo.bare?
-
-        else
         begin
-          @git.apply_patch(patch)
-        rescue Java::OrgEclipseJgitApiErrors::PatchApplyException, Java::OrgEclipseJgitApiErrors::PatchFormatException, Java::OrgEclipseJgitApiErrors::GitAPIException
+          applier = RJGit::Plumbing::ApplyPatchToIndex.new(@repo, patch)
+          applier.build_map
+        rescue Java::OrgEclipseJgitApiErrors::PatchApplyException, Java::OrgEclipseJgitApiErrors::PatchFormatException
```

Comment:
Same question as in the RJGit code: should this be a JGit exception or rather an RJGit exception?

Reply:
Went for an RJGit exception

Reply:
By which you mean, I think, that RJGit will throw an RJGit Exception, that will be rescued here in the adapter, correct?

Reply:
Indeed!